identity/conf/routes

```diff
@@ -17,6 +17,7 @@ POST        /password/change                        controllers.ChangePasswordCo
 GET         /password/reset-confirmation            controllers.ResetPasswordController.renderPasswordResetConfirmation(returnUrl: Option[String])
 GET         /password/email-sent                    controllers.ResetPasswordController.renderEmailSentConfirmation
 POST        /password/guest                         controllers.editprofile.EditProfileController.guestPasswordSet()
+POST        /actions/auth/ajax                      controllers.AuthenticationController.authenticateUsernamePassword()
```

Comment:
we should have a different root for cross-domain non-csrf actions perhaps, to tell them apart and treat them a bit more especially (a bit like a public API, we can't touch their behaviour). Maybe something like `external-actions/v1/auth-ajax`?

Reply:
I'm working on adding CSRF protection to this endpoint, at the moment just testing the core functionality.

Reply:
we can't have that because dotcom can't get the token :((

Reply:
You were right, so I just fixed the way the `Access-Control-Allow-Origin` is set instead, so this endpoint should only be able to be called from browsers, with valid origin domains.

Reply:
neat!!